flepimop/gempyor_pkg/src/gempyor/utils.py

```diff
@@ -100,6 +100,22 @@ def read_df(
 
 
 def command_safe_run(command, command_name="mycommand", fail_on_fail=True):
```

Comment:
I think type hints in the function definitions themselves instead of in the documentation would be much preferred. For example this would be something like:
```python
def command_safe_run(command: str, command_name: str = "mycommand", fail_on_fail: bool = True) -> tuple[int, str, str]:
    pass
```
or whatever the correct hints are (just guessing from glancing at this). Doing it this way has some advantages:

1. Makes type related portions of the documentation standard. I know exactly where to find the type for a param rather than having to read the full param documentation.
2. Allows for changing type definitions independent of documentation. i.e. Adding support for a new type.
3. Python's type hints integrate well with lots of IDEs already and this opens the door for using static analysis tools like [`mypy`](https://mypy-lang.org/) much further down the road.

Here is a good cheat sheet for getting started: https://mypy.readthedocs.io/en/latest/cheat_sheet_py3.html.